app/src/main/java/org/mozilla/reference/browser/components/Core.kt

```diff
@@ -141,7 +140,7 @@ class Core(private val context: Context) {
      * The storage component to persist browsing history (with the exception of
      * private sessions).
      */
-    val historyStorage: HistoryStorage by lazy { HistoryDatabase(context) }
+    val historyDatabase: HistoryDatabase by lazy { HistoryDatabase(context) }
```

Comment:
This is the problem. Do not rename this and do not say this type is real, stick with the interfaces